lvsfunc/util.py

```diff
@@ -102,6 +103,45 @@ def get_prop(frame: vs.VideoFrame, key: str, t: Type[T]) -> T:
     return prop
 
 
+def select_frames(clips: Union[vs.VideoNode, List[vs.VideoNode]],
+                  indicies: Union[List[int], List[Tuple[int, int]]]) -> vs.VideoNode:
+    """
+    Select frames from one or more clips at specified indices.
+
+    :param clips:   A clip or a list of clips to select the frames from
+    :param indexes: Indices of frames to select.
+
+                    Provide a list of indices for a single clip, or for multiple clips, a list of tuples in the
+                    form ``(clip_index, frame_index)``
+
+    :return:        The selected frames in a single clip
+    """
+    clips = [clips] if isinstance(clips, vs.VideoNode) else clips
+    for pos, index in enumerate(indicies):
+        if isinstance(index, int):
+            indicies[pos] = (0, index)
+        clip_idx, frame_idx = indicies[pos]
+        if not (0 <= clip_idx <= len(clips)):
+            raise IndexError(f"Clip index {clip_idx} out of range")
+        if not (0 <= frame_idx <= clips[clip_idx].num_frames):
+            raise IndexError(f"Frame index {frame_idx} out of range for clip {clip_idx}")
+
+    if clips[0].format is None:
+        raise ValueError("Only constant format clips are supported.")
```

Comment:
probably best to merge these checks and just report something like `All input clips must be the same constant format.`

Reply:
Sure, easily done